toolchains/xslt-M4/schema-gen/make-json-schema-metamap.xsl

```diff
@@ -193,9 +193,11 @@
         <xsl:variable name="enumerations" as="node()*">
             <xsl:apply-templates select="$decl/constraint/allowed-values"/>
         </xsl:variable>
+        <xsl:variable name="allow-other" select="$enumerations/@allow-other = 'yes'"/>
+        <xsl:variable name="any-or-all" select="if ($allow-other) then 'anyOf' else 'allOf'"/> 
         <xsl:choose>
             <xsl:when test="exists($enumerations)">
```

Comment:
This should target only when the target of the allowed values is the current metaschema context. This based on the feedback from @kylelaker.

```suggestion

            <xsl:when test="$allow-other and (constraint/allowed-values/@target = '.' or empty(constraint/allowed-values/@target))">
```